src/compose/service-manager.coffee

```diff
@@ -101,24 +101,25 @@ module.exports = class ServiceManager extends EventEmitter
 			return @docker.getContainer(existingService.containerId)
 		.catch NotFoundError, =>
 
-			conf = service.toContainerConfig()
-			nets = service.extraNetworksToJoin()
-
 			@config.get('name')
 			.then (deviceName) =>
+
 				service.environment['RESIN_DEVICE_NAME_AT_INIT'] = deviceName
 
+				conf = service.toContainerConfig()
+				nets = service.extraNetworksToJoin()
+
 				@logger.logSystemEvent(logTypes.installService, { service })
 				@reportNewStatus(mockContainerId, service, 'Installing')
 
 				@docker.createContainer(conf)
-			.tap (container) =>
-				service.containerId = container.id
+				.tap (container) =>
```

Comment:
I think you don't need to indent the `tap` here.

Reply:
it's because of the use of `nets` here: https://github.com/resin-io/resin-supervisor/pull/724/files#diff-e7a20470a15d19708363a4eb6965b966R119

Reply:
Ah yeah 👍